source/Server/IntegratedAuthentication/IntegratedAuthenticationHost.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.HttpSys;
+using Octopus.Diagnostics;
+using Octopus.Server.Extensibility.Authentication.DirectoryServices.Configuration;
+using Octopus.Server.Extensibility.Authentication.DirectoryServices.DirectoryServices;
+using Octopus.Server.Extensibility.Extensions.Infrastructure.Web;
+using Octopus.Server.Extensibility.HostServices.Web;
+
+namespace Octopus.Server.Extensibility.Authentication.DirectoryServices.IntegratedAuthentication
+{
+    class IntegratedAuthenticationHost : IExecuteWhenServerStarts
```

Comment:
```suggestion
    class IntegratedAuthenticationHost : IExecuteWhenWebServerStarts
```
These all seem specific to the Web Server starting up.